micrometer-core/src/main/java/io/micrometer/core/instrument/observation/DefaultMeterObservationHandler.java

```diff
@@ -83,6 +83,8 @@ public void onStart(Observation.Context context) {
     }
 
     @Override
+    // TODO decide what to do about context.getName being Nullable
+    @SuppressWarnings("NullAway")
```

Comment:
This is a problem where we're doing `Timer.builder(context.getName())` where the builder does not accept a null name - meters must have a non-null name. Ideally, a Context should always have a name. Perhaps we should have a default name when one isn't set?

Reply:
Let's figure this out in a separate issue before M1, possibly when we migrate the observation modules